etc/sync-function-template.js

```diff
@@ -796,14 +796,34 @@ function synctos(doc, oldDoc) {
 
   var theDocDefinition = docDefinitions[theDocType];
 
+  if (theDocDefinition.customActions && typeof(theDocDefinition.customActions.onTypeIdentificationSucceeded) === 'function') {
+    theDocDefinition.customActions.onTypeIdentificationSucceeded(doc, oldDoc);
+  }
+
   authorize(doc, oldDoc, theDocDefinition);
 
+  if (theDocDefinition.customActions && typeof(theDocDefinition.customActions.onAuthorizationSucceeded) === 'function') {
+    theDocDefinition.customActions.onAuthorizationSucceeded(doc, oldDoc);
+  }
```

Comment:
if you wanted to get super fancy you could write an abstraction of this function that would take only the function name and execute it if present and of type function.